src/SizedArray.jl

```diff
@@ -53,11 +61,11 @@ end
 @inline convert(::Type{SA}, sa::SA) where {SA<:SizedArray} = sa
 
 # Back to Array (unfortunately need both convert and construct to overide other methods)
-@inline Array(sa::SizedArray) = sa.data
+@inline Array(sa::SizedArray{S}) where {S} = sa.data
```

Comment:
Should copy though?  Also don't need the `where {S}` now.

Reply:
OK, I'll keep this change then. It should copy and `Array{T}` should also change the element type, right? But not reshape for now.

Reply:
I forgot that these constructors don't change element type, nevermind.

I've changed them so that they make copies. Is this the right set of changes for the next version of `StaticArrays.jl`?

Reply:
> should also change the element type, right

Hmm, good question actually. I think we'll fail to do the right thing for `convert`, because for non-matching eltype we will dispatch to the fallback `convert(::Array, ::StaticArray)`? I guess this has been a bug for a while.